### PR TITLE
Support using escaped reserved words in column names

### LINKED ISSUE #677: Support escaping identifier names.
In standard SQL we can use escaped names for tables and columns. We should support this in KSQL too.

### PR DESCRIPTION
**Pull request by hjafarpour**
This PR adds support for escaped column names. At the moment we support escaped column name only in `CREATE STREAM` and `CREATE TABLE` statements. We don't support renaming an existing column name with an escaped one using alias in the `CSAS`/`CTAS` statements.
User will be able to have create stream/table statements such as the following:

` CREATE STREAM orders ("group" bigint, orderid bigint, itemid varchar, orderunits double, arraycol array<double>, mapcol map<varchar, double>) WITH (kafka_topic = 'orders_topic_int_json' , value_format = 'json');`

The column name in the topic is not quoted and ksql serde classes will ignore quotes while deserializing messages from kafka topics.
The quoted columns can be used the same as the other columns. Here is an example:
`CREATE STREAM S1 AS SELECT " group" , "group"/1000 AS gr from orders;`.

Fixes #677

### CODE REVIEW

**Comment by apurvam** [comment]
Ha. I thought the plan was to print better error messages only for now :)

File: ksql-engine/src/main/java/io/confluent/ksql/QueryEngine.java [hunk 1]
```diff
@@ -235,7 +243,9 @@ StructuredDataSource getResultDatasource(final Select select, final String name)
         dataSource.schema(),
         null,
         null,
-        ksqlTopic
+        ksqlTopic,
+        quotedFieldNames
+
```
Comment (satybald) [inline ksql-engine/src/main/java/io/confluent/ksql/QueryEngine.java]: nit: new line

File: ksql-engine/src/main/java/io/confluent/ksql/physical/PhysicalPlanBuilder.java [hunk 2]
```diff
@@ -357,5 +361,18 @@ private KafkaStreams buildStreams(
     );
     return kafkaStreamsBuilder.buildKafkaStreams(builder, new StreamsConfig(newStreamsProperties));
   }
+
+  private Set<String> updateTheQuotedNameSet(Schema schema, Set<String> currentQuotedNames) {
+    Set<String> newQuotedNames = new HashSet<>();
+    Set<String> outputSchemaFieldNames =
+        schema.fields().stream().map(field -> field.name()).collect(Collectors.toSet());
+
+    for (String currentQuotedName: currentQuotedNames) {
```
Comment (satybald) [inline ksql-engine/src/main/java/io/confluent/ksql/physical/PhysicalPlanBuilder.java]: what are your thoughts to continue function style?
```
return currentQuotedNames
.filter(name -> outputSchemaFieldNames.contains(name))
.collect(Collectors.toSet())
```

**Comment by big-andy-coates** [comment]
#1050 is also about quoting things... might be worth checking changes / approach are consistent / compatible.

**Comment by hjafarpour** [comment]
Closing this so we can have more complete solution by @big-andy-coates.

**Pull request closed without merging**